suspect/io/twix.py

```diff
@@ -90,26 +90,55 @@ def calculate_orientation(normal):
     return "SAG"
 
 
-def read_double(name, header_string):
-    substring = re.search(r"<ParamDouble.\"{}\">  {{ <Precision> \d+(  -?[0-9\.]+)?  }}".format(name), header_string)
-    if not substring:
-        raise KeyError(r'ParamDouble."{}" not found in header string'.format(name))
-    number_string = substring.group(1)
-    return float(number_string) if number_string else 0
+def get_meta_regex(regex_list, header_string, convert=1, default=None):
+    """
+    Extract metadata using list of regex and apply unit convertion if value is number.
+    If regex found, but value is empty (usually empty string), will return default.
+
+    Parameters
+    ----------
+    regex_list : List of regex string
+    header_string : TWIX header string
+    convert : Unit convertion, if value is number. Defaults to 1 (means no convertion)
+    default : Default value if match found, but value is empty. Defaults to None.
+
+    Returns
+    -------
+    If number, converted to float and converted to unit (by multiplying by convert).
+    If non-number, return as is. If empty, return default.
+
+    Raises
+    ------
+    KeyError: if no matches found.
+    """
+    rgx_found = False
```

Comment:
I can't figure out what the purpose of `rgx_found` is, it looks like it is for terminating the loop, but then that is done with the `return` so this line and 118 below seem unnecessary?

Reply:
You're right! It was for the previous version before I updated this and I forgot to clean that up! Will remove this. Thanks for pointing this out. :-)